src/bson.c

```diff
@@ -131,8 +131,15 @@ static bool php_phongo_bson_visit_binary(const bson_iter_t *iter ARG_UNUSED, con
 
 static bool php_phongo_bson_visit_undefined(const bson_iter_t *iter, const char *key, void *data) /* {{{ */
 {
-	TSRMLS_FETCH();
-	phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "Detected unsupported BSON type 0x06 (undefined) for fieldname \"%s\"", key);
+	zval *retval = PHONGO_BSON_STATE_ZCHILD(data);
+
+	mongoc_log(MONGOC_LOG_LEVEL_WARNING, MONGOC_LOG_DOMAIN, "Detected unsupported BSON type 0x06 (undefined) for fieldname \"%s\"", key);
+
+	if (((php_phongo_bson_state *)data)->is_visiting_array) {
+		add_next_index_null(retval);
+	} else {
+		add_assoc_null(retval, key);
```

Comment:
In the interest of restoring 1.2's behavior, do you think it'd be better to leave the fields unset? That's less chance that the user mixes one of these up with a real BSON `null` type.

When we actually introduce the classes for these deprecated types, then we can start adding them to the PHP HashTable.

Reply:
Fixed